.golangci.yml

```diff
@@ -0,0 +1,140 @@
+linters-settings:
+  depguard:
+    list-type: blacklist
+    packages:
+      - github.com/jenkins-x/jx/v2/pkg/log/
+    packages-with-error-message:
+      - github.com/jenkins-x/jx/v2/pkg/log/: "use jenkins-x/jx-logging instead"
+  dupl:
+    threshold: 100
+  exhaustive:
+    default-signifies-exhaustive: false
+  funlen:
+    lines: 200
+    statements: 150
+  goconst:
+    min-len: 3
+    min-occurrences: 3
+  gocritic:
+    enabled-tags:
+      - diagnostic
+      - experimental
+      - opinionated
+      - performance
+      - style
+    disabled-checks:
+      - dupImport # https://github.com/go-critic/go-critic/issues/845
+      - ifElseChain
+      - octalLiteral
+      - whyNoLint
+      - wrapperFunc
+  gocyclo:
+    min-complexity: 15
+  goimports:
+  golint:
+    min-confidence: 0
+  gofmt:
+    simplify: true
+  gomnd:
+    settings:
+      mnd:
+        # don't include the "operation" and "assign"
+        checks: argument,case,condition,return
+  govet:
+    check-shadowing: true
+    settings:
+      printf:
+        funcs:
+          - (github.com/jenkins-x/jx-logging/pkg/log/Logger()).Debugf
+          - (github.com/jenkins-x/jx-logging/pkg/log/Logger()).Infof
+          - (github.com/jenkins-x/jx-logging/pkg/log/Logger()).Warnf
+          - (github.com/jenkins-x/jx-logging/pkg/log/Logger()).Errorf
+          - (github.com/jenkins-x/jx-logging/pkg/log/Logger()).Fatalf
+  lll:
+    line-length: 140
+  maligned:
+    suggest-new: true
+  misspell:
+  nolintlint:
+    allow-leading-space: true # don't require machine-readable nolint directives (i.e. with no leading space)
+    allow-unused: false # report any unused nolint directives
+    require-explanation: false # don't require an explanation for nolint directives
+    require-specific: false # don't require nolint directives to be specific about which linter is being skipped
+linters:
+  # please, do not use `enable-all`: it's deprecated and will be removed soon.
+  # inverted configuration with `enable-all` and `disable` is not scalable during updates of golangci-lint
+  disable-all: true
+  enable:
+    - asciicheck
+#    - bodyclose
+    - deadcode
+    - depguard
+#    - dogsled
+#    - dupl
+    - errcheck
+#    - funlen
+#    - goconst
+    - gofmt
+    - goimports
+    - goprintffuncname
+    - gosec
+#    - gosimple
+#    - ineffassign
+    - interfacer
```

Comment:
Interfacer is deprecated, and not recommended for real code anymore, so may be it should not be enabled? https://github.com/mvdan/interfacer

Reply:
Interesting, I didn't realise this.  I can remove in another PR (this was already enabled on the existing linter config)